src/dmd/func.d

```diff
@@ -152,13 +152,48 @@ enum FUNCFLAG : uint
     compileTimeOnly  = 0x100,  /// is a compile time only function; no code will be generated for it
 }
 
+/***********************************************************
+ * Tuple of result identifier (possibly null) and statement.
+ * This is used to store out contracts: out(id){ ensure }
+ */
+extern (C++) struct Ensure
+{
+    Identifier id;
+    Statement ensure;
+
+    Ensure syntaxCopy()
+    {
+        return Ensure(id, ensure.syntaxCopy());
+    }
+
+    /*****************************************
+     * Do syntax copy of an array of Ensure's.
+     */
+    static Ensures* arraySyntaxCopy(Ensures* a)
+    {
+        Ensures* b = null;
+        if (a)
+        {
+            b = a.copy();
+            foreach(i,e;*a)
```

Comment:
Add spaces in foreach (`i, e; *a`) - check for this style everywhere as I've seen it more than once.